kurobako_core/src/problem.rs

```diff
@@ -64,8 +66,16 @@ impl ProblemSpecBuilder {
         self
     }
 
+    /// Sets the value reference point of this problem.
+    pub fn reference_point(mut self, reference_point: Option<Params>) -> Self {
+        self.reference_point = reference_point;
+        self
+    }
+
     /// Builds a `ProblemSpec` with the given settings.
     pub fn finish(self) -> Result<ProblemSpec> {
+        self.validate()?;
```

Comment:
Please wrap this expression by the `track!` macro (it adds trace information when an error occurs).
```suggestion
        track!(self.validate())?;
```